gtsam/slam/SmartFactorBase.h

```diff
@@ -207,10 +207,17 @@ class SmartFactorBase: public NonlinearFactor {
     Vector ue = cameras.reprojectionError(point, measured_, Fs, E);
     if (body_P_sensor_ && Fs) {
       const Pose3 sensor_P_body = body_P_sensor_->inverse();
+      size_t camera_dim = size_t(traits<CAMERA>::dimension);
+      size_t pose_dim = size_t(traits<Pose3>::dimension);
+
       for (size_t i = 0; i < Fs->size(); i++) {
-        const Pose3 w_Pose_body = cameras[i].pose() * sensor_P_body;
-        Matrix J(6, 6);
-        const Pose3 world_P_body = w_Pose_body.compose(*body_P_sensor_, J);
+        const Pose3 world_P_body = cameras[i].pose() * sensor_P_body;
+        Matrix J = Matrix::Zero(camera_dim, camera_dim);
```

Comment:
Why switch to dynamic? Make static, use setZero(), save a malloc

Reply:
Gotcha. It was already dynamic for some reason so was trying not to inadvertently introduce any bugs.